src/msix/pack/XmlWriter.cpp

```diff
@@ -0,0 +1,83 @@
+//
+//  Copyright (C) 2019 Microsoft.  All rights reserved.
+//  See LICENSE file in the project root for full license information.
+//
+#include "XmlWriter.hpp"
+#include "StringStream.hpp"
+#include "ComHelper.hpp"
+#include "Exceptions.hpp"
+#include "MsixErrors.hpp"
+
+#include <stack>
+#include <string>
+
+namespace MSIX {
+
+        const static char* xmlStart = "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"true\"?><";
+
+        // Adds xml header declaration plus the name of the root element
+        XmlWriter::XmlWriter(const std::string& root)
+        {
+            m_elements.emplace(root);
+            m_stream = ComPtr<IStream>::Make<StringStream>();
+            std::string start = xmlStart + root;
+            ULONG copy;
+            ThrowHrIfFailed(m_stream->Write(static_cast<const void*>(start.data()), static_cast<ULONG>(start.size()), &copy));

```

Comment:
>Write(static_cast<const void*>(start.data()), static_cast<ULONG>(start.size()), &copy) [](start = 38, length = 86)

RE: that standalone helper, this could just be:

WriteString(start)